src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -402,4 +446,36 @@ public Command getAutonomousCommandFromTrajectory(Trajectory trajectory) {
             this))
         .andThen(this::stopDrive, this);
   }
+
+  public void shiftLow() {
+    shifter.set(Value.kForward);
+    shifterState = ShifterState.low;
+  }
+
+  public void shiftHigh() {
+    shifter.set(Value.kReverse);
+    shifterState = ShifterState.high;
+  }
+
+  public ShifterState getShifterState() {
+    return shifterState;
+  }
+
+  public void toggleShift() {
+    if (shifterState == ShifterState.high) {
+      shiftLow();
+    } else if (shifterState == ShifterState.low) {
+      shiftHigh();
+    } else {
+      throw new IllegalStateException("bruh");
+    }
+  }
+
+  private boolean isLowGearOptimal() {
+    if ((getLeftEncoderMetersPerSecondVelocity() + getRightEncoderMetersPerSecondVelocity()) / 2
```

Comment:
if Math.abs(average) < shiftPointsMetersPerSecond

also
if difference between left side and right side velocity > 20% don't shift. that prevents you from shifting into low gear while turning.
or
do we want it to shift to high automatically when turning?